jwst/tweakreg/tweakreg_step.py

```diff
@@ -614,31 +616,35 @@ def _imodel2wcsim(self, image_model):
 
 
 def _common_name(group, all_group_names=None):
-    file_names = [path.splitext(im.meta.filename)[0].strip('_- ')
-                  for im in group]
+    file_names = [
+        path.splitext(im.meta.filename)[0].strip('_- ') for im in group
+    ]
 
     cn = path.commonprefix(file_names)
 
     if all_group_names is None:
-        if not cn:
-            return 'Unnamed Group'
+        if cn:
+            return cn
+        else:
+            _common_name.gid += 1
+            return f"Group #{_common_name.gid}"
     else:
-        if not cn or cn in all_group_names:
-            # find the smallest group number to make "Group #..." unique
-            max_id = 1
-            if not cn:
-                cn = "Group #"
-            for name in all_group_names:
-                try:
-                    cid = int(name.lstrip(cn))
-                    if cid >= max_id:
-                        max_id = cid + 1
-                except ValueError:
-                    pass
-            cn = f"{cn}{max_id}"
-        all_group_names.append(cn)
-
-    return cn
+        if not cn:
+            cn = "Group"
+
+        grp_no = all_group_names.get(cn, 0) + 1
+        grp_id = f"{cn} #{grp_no}"
+        if grp_id in all_group_names:
+            # just try some shortened uuid until it is unique:
+            while grp_id in all_group_names:
+                grp_id = f"{cn}_{str(uuid.uuid4())[-6:]}"
+            all_group_names[grp_id] = 0
+        else:
+            all_group_names[cn] = grp_no
+        return grp_id
+
+
+_common_name.gid = 0
```

Comment:
Can we initialize `gid` as a default argument in the function call (_i.e._ `def _common_name(group, all_group_names=None, gid=0):`)? I think strengthening encapsulation would improve flexibility and reusability.

Reply:
@mairanteodoro Sorry, but after discussing this offline with @braingram , we decided to remove this function altogether. It looks like now `model.meta.group_id` is guaranteed to be set if `model_container.models_grouped` property has been invoked (which is the case for `tweakreg_step`).